sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -174,13 +174,21 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
                 hof.failAnalysis(
                   s"cannot resolve '${hof.sql}' due to argument data type mismatch: $message")
             }
+          // Check if there are expressions with children containing star expressions.
+          case e if e.children.exists(_.isInstanceOf[Star]) =>
+            e.failAnalysis(s"Invalid usage of '*' in expression '${e.prettyName}'")
         }
 
         operator transformExpressionsUp {
           case a: Attribute if !a.resolved =>
             val from = operator.inputSet.toSeq.map(_.qualifiedName).mkString(", ")
             a.failAnalysis(s"cannot resolve '${a.sql}' given input columns: [$from]")
 
+          case s @ UnresolvedStar(Some(target)) =>
```

Comment:
We still need this error check even if we check it in L178?

Reply:
Yup L178 checks invalid star usage inside expressions first, for example, `sum(t.*)` and here the unresolved star is not inside any expressions.